kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,606 @@
+  otherServersSize=${#otherServers[@]}
+  if [ "${otherServersSize}" -gt 0 ]; then
+    for otherServer in "${otherServers[@]}"; do
+      sortedByAlwaysServers+=($otherServer)
+    done
+  fi
+}
+
+#
+# Get replica count for a cluster
+# $1 - Domain resource in json format
+# $2 - Name of cluster 
+# $3 - Return value containing replica count
+#
+function getReplicaCount {
+  local domainJson=$1
+  local clusterName=$2
+  local __replicaCount=$3
+
+  replicasCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).replicas"
+  replicaCount=$(echo ${domainJson} | jq "${replicasCmd}")
+  eval $__replicaCount="'${replicaCount}'"
+
+}
+
+#
+# Check servers started in a cluster based on server start policy and 
+# replica count.
+# $1 - Domain resource in json format
+# $2 - Name of server 
+# $3 - Name of cluster 
+# $4 - Indicates if replicas will stay constant, incremented or decremented
+# $5 - Indicates if policy of current server will stay same or unset
+# $6 - Return value indicating if current server will be started
+#
+function checkStartedServers {
+  local domainJson=$1
+  local serverName=$2
+  local clusterName=$3
+  local withReplicas=$4
+  local withPolicy=$5
+  local __started=$6
+  local localServerName=""
+  local policy=""
+  local replicaCount=0
+  local currentReplicas=0
+  local startedServers=()
+  local sortedByAlwaysServers=()
+  
+  # Get sorted list of servers in 'sortedByAlwaysServers' array
+  getSortedListOfServers "${domainJson}" "${serverName}" "${clusterName}" "${withPolicy}"
+  getReplicaCount "${domainJson}" "${clusterName}" replicaCount
+  # Increment or decrement the replica count based on 'withReplicas' input parameter
+  if [ "${withReplicas}" == "INCREASED" ]; then
+    replicaCount=$((replicaCount+1))
+  elif [ "${withReplicas}" == "DECREASED" ]; then
+    replicaCount=$((replicaCount-1))
+  fi
+  sortedByAlwaysSize=${#sortedByAlwaysServers[@]}
+  if [ "${sortedByAlwaysSize}" -gt 0 ]; then
+    for localServerName in "${sortedByAlwaysServers[@]}"; do
+      getEffectivePolicy "${domainJson}" "${localServerName}" "${clusterName}" policy
+      # Update policy when server name matches current server and unsetting
+      if [[ "${serverName}" == "${localServerName}" && "${withPolicy}" == "UNSET" ]]; then
+        policy=UNSET
+      fi
+      # check if server should start based on replica count, policy and current replicas
+      shouldStart "${currentReplicas}" "${policy}" "${replicaCount}" result
+      if [ "${result}" == 'True' ]; then
+        # server should start, increment current replicas and add server to list of started servers
+        currentReplicas=$((currentReplicas+1))
+        startedServers+=(${localServerName})
+      fi
+    done
+  fi
+  startedSize=${#startedServers[@]}
+  if [ ${startedSize} -gt 0 ]; then
+    # check if current server is in the list of started servers
+    if checkStringInArray ${serverName} ${startedServers[@]}; then
+      eval $__started="true"
```

Comment:
In some parts of the code "true/false" is used, while others "True/False" is used - can you standardize on the former?

Reply:
or maybe standardize on "TRUE/FALSE" (your choice) - main point is to use same values throughout the scripts

Reply:
Changed to "true/false".